File: apps/sdkwork-terminal-flutter-mobile/lib/bootstrap/sdk_clients.dart
import 'iam_runtime.dart';
import 'runtime.dart';

class SdkClients {
  final String baseUrl;
  final String apiVersion;
  final IamRuntime iamRuntime;

  SdkClients({
    required this.baseUrl,
    required this.apiVersion,
    required this.iamRuntime,
  });

  factory SdkClients.create() {
    final runtimeConfig = RuntimeConfig.fromEnvironment();
    final iamRuntime = getIamRuntime();

    return SdkClients(
      baseUrl: runtimeConfig.baseUrl,
      apiVersion: runtimeConfig.apiVersion,
      iamRuntime: iamRuntime,
    );
  }
}

SdkClients? _cachedSdkClients;

SdkClients getSdkClients() {
  return _cachedSdkClients ??= SdkClients.create();
}

File: apps/sdkwork-terminal-pc/packages/sdkwork-terminal-pc-core/src/bootstrap/sdkClients.ts
import type { SdkworkAppClient } from '@sdkwork/iam-app-sdk';
import type { AuthTokenManager } from '@sdkwork/sdk-common';

import { getIamComposition, getIamRuntime } from './iamRuntime';
import { getRuntimeConfig } from './runtime';

export interface SdkClients {
  baseUrl: string;
  apiVersion: string;
  appbaseApp: SdkworkAppClient;
  tokenManager: AuthTokenManager;
}

let cachedSdkClients: SdkClients | null = null;

export function createSdkClients(): SdkClients {
  const config = getRuntimeConfig();
  const iam = getIamRuntime();

  return {
    baseUrl: config.baseUrl,
    apiVersion: config.apiVersion,
    appbaseApp: iam.appbaseApp,
    tokenManager: iam.tokenManager,
  };
}

export function getSdkClients(): SdkClients {
  if (!cachedSdkClients) {
    cachedSdkClients = createSdkClients();
  }
  return cachedSdkClients;
}

export function getAppbaseAppClient(): SdkworkAppClient {
  return getIamComposition().appbaseApp;
}
